elki/src/main/java/de/lmu/ifi/dbs/elki/datasource/parser/ClusteringVectorParser.java

```diff
@@ -152,7 +152,7 @@ public Event nextEvent() {
       return ret;
     }
     try {
-      while(reader.nextLineExceptComments()) {
+      if(reader.nextLineExceptComments()) {
```

Comment:
I know that it may contain more than one result, but there is no way to get out of the `while`-loop without either returning a value or throwing an exception. Therefore, a next iteration of the `while`-loop will never be executed. I think the following lines are read by calling this method multiple times. 

As it currently stands, the `while` part is confusing in my opinion as it does not actually loop.